targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC55S69/TARGET_M33_NS/device/TOOLCHAIN_GCC_ARM/LPC55S69_cm33_core0_flash.ld

```diff
@@ -57,15 +57,19 @@ __ram_vector_table__ = 1;
     #define MBED_BOOT_STACK_SIZE 0x400
 #endif
 
+#if !defined(MBED_NVIC_SIZE)
+    #define MBED_NVIC_SIZE 0x130
```

Comment:
This should probably be defined as NVIC_NUM_VECTORS * 4 to avoid magic numbers.

See https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC55S69/TARGET_M33_NS/device/cmsis_nvic.h#L43

Reply:
923c1f2

Reply:
Excellent. Thanks

Reply:
There is an 8-byte alignment requirement, if I recall correctly, so just using `NVIC_NUM_VECTORS * 4` may not quite be magic enough in general. Not sure if GCC has a handy alignment built-in to help you here. I know the ARM linker would let you use `AlignExpr(NVIC_NUM_VECTORS * 4,8)`.

Reply:
> There is an 8-byte alignment requirement, if I recall correctly, so just using `NVIC_NUM_VECTORS * 4` may not quite be magic enough in general. Not sure if GCC has a handy alignment built-in to help you here. I know the ARM linker would let you use `AlignExpr(NVIC_NUM_VECTORS * 4,8)`.
> 
> Sure, it happens to be aligned on this platform, but if someone were to copy the general-looking implementation to another...

@kjbracey-arm 
The mention of the image start address alignment I saw in the [user manual](https://www.nxp.com/docs/en/user-guide/UM11126.pdf) is a requirement for a 32-bit word alignment (see linked user manual on page 153).
Should it not be `AlignExpr(NVIC_NUM_VECTORS * 4, 4)` instead for the ARM toolchain?

Reply:
Individual chips might permit things to be 4-byte aligned, but architecturally regions are supposed to be 8-byte aligned these days. The ARMC6 toolchain is only letting you get away with 4-byte alignment because `--legacyalign` is being passed.

There are certain instructions that only work or work faster with 8-byte alignment that the compiler may wish to use.

See other PRs #8013 and #10373, correcting platforms to 8-byte alignment.

Reply:
Thank you @kjbracey-arm I will modify the ARM toolchain linker file. However, for GCC I believe it is handled with `. = ALIGN(8)` which is already present in the section definitions.

Reply:
Yes, that probably works. Crosscheck how other implementations do it, including in those PRs, there are a couple of common patterns (including hard-coding alignment to 8 by hand in quite a few).

Reply:
https://github.com/ARMmbed/mbed-os/compare/923c1f21ffed881e36bf90310ce14b2bf0d35401..b0804c4a0d6bc838e33334573c71e274028cb90b